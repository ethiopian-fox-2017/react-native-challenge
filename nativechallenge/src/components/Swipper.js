import React, { Component } from 'react';
import { Image } from 'react-native';

import {
  Container,
  Icon,
  View,
  DeckSwiper,
  Card,
  CardItem,
  Thumbnail,
  Text,
  Left,
  Body } from 'native-base';

  const cards = [
    {
        text: 'Card One',
        name: 'Salad Healthy',
        image: 'https://media.istockphoto.com/photos/pumpkin-salad-picture-id635912088',
    },
    {
        text: 'Card Two',
        name: 'Mayonais',
        image: 'https://media.istockphoto.com/photos/delicious-benedict-eggs-picture-id610847948',
    },
    {
        text: 'Card Three',
        name: 'Selai Nanas',
        image: 'https://media.istockphoto.com/photos/egg-benedict-toast-english-breakfast-plate-concept-picture-id483281364',
    },
    {
        text: 'Card Four',
        name: 'Cafe Latte',
        image: 'https://media.istockphoto.com/photos/mushroom-cream-soup-isolated-on-white-picture-id518620350',
    },
    {
        text: 'Card Five',
        name: 'Sledri',
        image: 'https://media.istockphoto.com/photos/pumpkin-salad-picture-id635912088',
    },

];

class Swipper extends React.Component {
  render () {
    return (
      <Container>
        <View>
          <DeckSwiper
            dataSource={cards}
            renderItem={(item) =>
                <Card style={{ elevation: 4 }}>
                  <CardItem>
                    <Left>
                      <Thumbnail source={{ uri: item.image }} style={{ height: 200, width: 200 }} />
                      <Body>
                        <Text>{item.text}</Text>
                        <Text note>{item.name}</Text>
                      </Body>
                    </Left>
                  </CardItem>
                    <CardItem cardBody>
                        <Text>{ item.name }</Text>
                    </CardItem>
                    <CardItem>
                        <Icon name="heart" style={{ color: '#ED4A6A' }} />
                        <Text>{item.name}</Text>
                    </CardItem>
                </Card>
              }
            />
        </View>
      </Container>
    );
  }
}

export default Swipper;
