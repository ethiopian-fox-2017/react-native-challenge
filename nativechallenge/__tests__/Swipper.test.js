import 'react-native';
import React from 'react';
import Swipper from '../src/components/Swipper.js';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

it('renders correctly', () => {
  const tree = shallow(
    <Swipper />
  );
  expect(tree).toMatchSnapshot()
});
