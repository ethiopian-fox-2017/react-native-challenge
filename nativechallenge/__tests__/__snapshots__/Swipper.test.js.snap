// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Styled(Container)
    virtual={undefined}
>
    <Styled(ViewNB)
        virtual={undefined}
    >
        <Styled(DeckSwiper)
            dataSource={
                Array [
                    Object {
                      "image": "https://media.istockphoto.com/photos/pumpkin-salad-picture-id635912088",
                      "name": "Salad Healthy",
                      "text": "Card One",
                    },
                    Object {
                      "image": "https://media.istockphoto.com/photos/delicious-benedict-eggs-picture-id610847948",
                      "name": "Mayonais",
                      "text": "Card Two",
                    },
                    Object {
                      "image": "https://media.istockphoto.com/photos/egg-benedict-toast-english-breakfast-plate-concept-picture-id483281364",
                      "name": "Selai Nanas",
                      "text": "Card Three",
                    },
                    Object {
                      "image": "https://media.istockphoto.com/photos/mushroom-cream-soup-isolated-on-white-picture-id518620350",
                      "name": "Cafe Latte",
                      "text": "Card Four",
                    },
                    Object {
                      "image": "https://media.istockphoto.com/photos/pumpkin-salad-picture-id635912088",
                      "name": "Sledri",
                      "text": "Card Five",
                    },
                  ]
            }
            renderItem={[Function]}
            virtual={undefined}
        />
    </Styled(ViewNB)>
</Styled(Container)>,
  "nodes": Array [
    <Styled(Container)
      virtual={undefined}
>
      <Styled(ViewNB)
            virtual={undefined}
      >
            <Styled(DeckSwiper)
                  dataSource={
                        Array [
                              Object {
                                "image": "https://media.istockphoto.com/photos/pumpkin-salad-picture-id635912088",
                                "name": "Salad Healthy",
                                "text": "Card One",
                              },
                              Object {
                                "image": "https://media.istockphoto.com/photos/delicious-benedict-eggs-picture-id610847948",
                                "name": "Mayonais",
                                "text": "Card Two",
                              },
                              Object {
                                "image": "https://media.istockphoto.com/photos/egg-benedict-toast-english-breakfast-plate-concept-picture-id483281364",
                                "name": "Selai Nanas",
                                "text": "Card Three",
                              },
                              Object {
                                "image": "https://media.istockphoto.com/photos/mushroom-cream-soup-isolated-on-white-picture-id518620350",
                                "name": "Cafe Latte",
                                "text": "Card Four",
                              },
                              Object {
                                "image": "https://media.istockphoto.com/photos/pumpkin-salad-picture-id635912088",
                                "name": "Sledri",
                                "text": "Card Five",
                              },
                            ]
                  }
                  renderItem={[Function]}
                  virtual={undefined}
            />
      </Styled(ViewNB)>
</Styled(Container)>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Swipper />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Swipper {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Styled(Container)
          virtual={undefined}
>
          <Styled(ViewNB)
                    virtual={undefined}
          >
                    <Styled(DeckSwiper)
                              dataSource={
                                        Array [
                                                  Object {
                                                    "image": "https://media.istockphoto.com/photos/pumpkin-salad-picture-id635912088",
                                                    "name": "Salad Healthy",
                                                    "text": "Card One",
                                                  },
                                                  Object {
                                                    "image": "https://media.istockphoto.com/photos/delicious-benedict-eggs-picture-id610847948",
                                                    "name": "Mayonais",
                                                    "text": "Card Two",
                                                  },
                                                  Object {
                                                    "image": "https://media.istockphoto.com/photos/egg-benedict-toast-english-breakfast-plate-concept-picture-id483281364",
                                                    "name": "Selai Nanas",
                                                    "text": "Card Three",
                                                  },
                                                  Object {
                                                    "image": "https://media.istockphoto.com/photos/mushroom-cream-soup-isolated-on-white-picture-id518620350",
                                                    "name": "Cafe Latte",
                                                    "text": "Card Four",
                                                  },
                                                  Object {
                                                    "image": "https://media.istockphoto.com/photos/pumpkin-salad-picture-id635912088",
                                                    "name": "Sledri",
                                                    "text": "Card Five",
                                                  },
                                                ]
                              }
                              renderItem={[Function]}
                              virtual={undefined}
                    />
          </Styled(ViewNB)>
</Styled(Container)>,
        "_debugID": 2,
        "_renderedOutput": <Styled(Container)
          virtual={undefined}
>
          <Styled(ViewNB)
                    virtual={undefined}
          >
                    <Styled(DeckSwiper)
                              dataSource={
                                        Array [
                                                  Object {
                                                    "image": "https://media.istockphoto.com/photos/pumpkin-salad-picture-id635912088",
                                                    "name": "Salad Healthy",
                                                    "text": "Card One",
                                                  },
                                                  Object {
                                                    "image": "https://media.istockphoto.com/photos/delicious-benedict-eggs-picture-id610847948",
                                                    "name": "Mayonais",
                                                    "text": "Card Two",
                                                  },
                                                  Object {
                                                    "image": "https://media.istockphoto.com/photos/egg-benedict-toast-english-breakfast-plate-concept-picture-id483281364",
                                                    "name": "Selai Nanas",
                                                    "text": "Card Three",
                                                  },
                                                  Object {
                                                    "image": "https://media.istockphoto.com/photos/mushroom-cream-soup-isolated-on-white-picture-id518620350",
                                                    "name": "Cafe Latte",
                                                    "text": "Card Four",
                                                  },
                                                  Object {
                                                    "image": "https://media.istockphoto.com/photos/pumpkin-salad-picture-id635912088",
                                                    "name": "Sledri",
                                                    "text": "Card Five",
                                                  },
                                                ]
                              }
                              renderItem={[Function]}
                              virtual={undefined}
                    />
          </Styled(ViewNB)>
</Styled(Container)>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Swipper />,
}
`;
